src/app/core/components/header/header.component.html

```diff
@@ -1,4 +1,4 @@
-<header class="header-container">
+<header class="header">
   <app-logo></app-logo>
-  <app-user-panel></app-user-panel>
+  <app-user-panel *ngIf="showUserPanel$ | async" [userName$]="userName$"></app-user-panel>
```

Comment:
I'm pretty sure that this logic must be within app-user-panel as a smart component.

Reply:
Okay